File: web/src/api/testOrder.js
import service from '@/utils/request'
import { ElMessage } from 'element-plus'

export const createOrder = (data) => {
  return service({
    url: '/testOrder/createOrder',
    method: 'post',
    data
  })
}

export const deleteOrder = (data) => {
  return service({
    url: '/testOrder/deleteOrder',
    method: 'delete',
    data
  })
}

export const deleteOrderByIds = (data) => {
  return service({
    url: '/testOrder/deleteOrderByIds',
    method: 'delete',
    data
  })
}

export const updateOrder = (data) => {
  return service({
    url: '/testOrder/updateOrder',
    method: 'put',
    data
  })
}

export const findOrder = (params) => {
  return service({
    url: '/testOrder/findOrder',
    method: 'get',
    params
  })
}

export const getOrderList = (params) => {
  return service({
    url: '/testOrder/getOrderList',
    method: 'get',
    params
  })
}

export const approveApply = (data) => {
  return service({
    url: '/testOrder/approveApply',
    method: 'put',
    data
  })
}

export const rejectApply = (data) => {
  return service({
    url: '/testOrder/rejectApply',
    method: 'put',
    data
  })
}

export const approveDelay = (data) => {
  return service({
    url: '/testOrder/approveDelay',
    method: 'put',
    data
  })
}

export const rejectDelay = (data) => {
  return service({
    url: '/testOrder/rejectDelay',
    method: 'put',
    data
  })
}

export const approveRefund = (data) => {
  return service({
    url: '/testOrder/approveRefund',
    method: 'put',
    data
  })
}

export const rejectRefund = (data) => {
  return service({
    url: '/testOrder/rejectRefund',
    method: 'put',
    data
  })
}

export const approveClaim = (data) => {
  return service({
    url: '/testOrder/approveClaim',
    method: 'put',
    data
  })
}

export const rejectClaim = (data) => {
  return service({
    url: '/testOrder/rejectClaim',
    method: 'put',
    data
  })
}

export const getOrderStatisticData = (params) => {
  return service({
    url: '/testOrder/getOrderStatisticData',
    method: 'get',
    params
  })
}

export const downloadExcel = (params) => {
  return service({
    url: '/testOrder/exportExcel',
    method: 'get',
    params: params,
    responseType: 'blob'
  }).then((res) => {
    handleExcelError(res)
  })
}

const handleExcelError = (res) => {
  if (typeof (res.data) !== 'undefined') {
    if (res.data.type === 'application/json') {
      const reader = new FileReader()
      reader.onload = function() {
        const message = JSON.parse(reader.result).msg
        ElMessage({
          showClose: true,
          message: message,
          type: 'error'
        })
      }
      reader.readAsText(new Blob([res.data]))
    }
  } else {
    const downloadUrl = window.URL.createObjectURL(new Blob([res]))
    const a = document.createElement('a')
    a.style.display = 'none'
    a.href = downloadUrl
    a.download = (new Date().getTime()).toString() + '.xlsx'
    a.click()
  }
}

File: server/router/lgjx/testOrder.go
package lgjx

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type TestOrderRouter struct {
}

func (s *TestOrderRouter) InitOrderRouter(Router *gin.RouterGroup) {
	testOrderRouter := Router.Group("testOrder").Use(middleware.OperationRecord())
	testOrderRouterWithoutRecord := Router.Group("testOrder")
	var testOrderApi = v1.ApiGroupApp.LgjxTestApiGroup.TestOrderApi
	{
		testOrderRouter.POST("createOrder", testOrderApi.CreateOrder)             // 新建Order
		testOrderRouter.DELETE("deleteOrder", testOrderApi.DeleteOrder)           // 删除Order
		testOrderRouter.DELETE("deleteOrderByIds", testOrderApi.DeleteOrderByIds) // 批量删除Order
		testOrderRouter.PUT("approveApply", testOrderApi.ApproveApply)            // 审批同意申请
		testOrderRouter.PUT("rejectApply", testOrderApi.RejectApply)              // 审批拒绝申请
		testOrderRouter.PUT("approveDelay", testOrderApi.ApproveDelay)            // 审批同意延期
		testOrderRouter.PUT("rejectDelay", testOrderApi.RejectDelay)              // 审批拒绝延期
		testOrderRouter.PUT("approveRefund", testOrderApi.ApproveRefund)          // 审批同意退函
		testOrderRouter.PUT("rejectRefund", testOrderApi.RejectRefund)            // 审批拒绝退函
		testOrderRouter.PUT("approveClaim", testOrderApi.ApproveClaim)            // 审批同意理赔
		testOrderRouter.PUT("rejectClaim", testOrderApi.RejectClaim)              // 审批拒绝理赔
		testOrderRouter.PUT("openLetter", testOrderApi.OpenLetter)                // 提交开函申请
	}
	{
		testOrderRouterWithoutRecord.GET("findOrder", testOrderApi.FindOrder)                         // 根据ID获取Order
		testOrderRouterWithoutRecord.GET("getOrderList", testOrderApi.GetOrderList)                   // 获取Order列表
		testOrderRouterWithoutRecord.GET("getOrderStatisticData", testOrderApi.GetOrderStatisticData) // 获取Order统计数据
		testOrderRouterWithoutRecord.GET("exportExcel", testOrderApi.ExportExcel)                     // 导出Order数据到excel
	}
}
